com.unity.netcode.gameobjects/Prototyping/NetworkTransform.cs

```diff
@@ -356,18 +502,146 @@ private void OnNetworkStateChanged(NetworkState oldState, NetworkState newState)
                 return;
             }
 
-            ApplyNetworkState(newState);
+            if (IsServer)
+            {
+                return; // todo use authority
+            }
+
+            var sentTime = new NetworkTime(NetworkManager.Singleton.ServerTime.TickRate, newState.SentTime);
+
+            if (newState.HasPositionX)
+            {
+                PositionXInterpolator.AddMeasurement(newState.PositionX, sentTime);
+            }
+
+            if (newState.HasPositionY)
+            {
+                PositionYInterpolator.AddMeasurement(newState.PositionY, sentTime);
+            }
+
+            if (newState.HasPositionZ)
+            {
+                PositionZInterpolator.AddMeasurement(newState.PositionZ, sentTime);
+            }
+
+            RotationInterpolator.AddMeasurement(Quaternion.Euler(newState.Rotation), sentTime);
+
+            if (newState.HasScaleX)
+            {
+                ScaleXInterpolator.AddMeasurement(newState.ScaleX, sentTime);
+            }
+
+            if (newState.HasScaleY)
+            {
+                ScaleYInterpolator.AddMeasurement(newState.ScaleY, sentTime);
+            }
+
+            if (newState.HasScaleZ)
+            {
+                ScaleZInterpolator.AddMeasurement(newState.ScaleZ, sentTime);
+            }
+
+            if (NetworkManager.Singleton.LogLevel == LogLevel.Developer)
+            {
+                var pos = new Vector3(newState.PositionX, newState.PositionY, newState.PositionZ);
+                Debug.DrawLine(pos, pos + Vector3.up + Vector3.left * Random.Range(0.5f, 2f), Color.green, k_DebugDrawLineTime, false);
+            }
```

Comment:
I'm not sure if we should reinterpret logging level as debugging level across the SDK — might be OK, we need to call this out.

Reply:
wanna start a thread? cause on my side, not sure I see what you mean. I would never use Netcode for GO's Developer log level in a live environment, it's way too noisy and takes way too much perf with all those string allocs

Reply:
bump